xdsl/ir/core.py

```diff
@@ -478,13 +478,13 @@ def remove_use(self, use: Use):
 
 
 @dataclass(eq=False)
-class SSAValue(IRWithUses, ABC):
+class SSAValue(Generic[AttributeCovT], IRWithUses, ABC):
     """
     A reference to an SSA variable.
     An SSA variable is either an operation result, or a basic block argument.
     """
 
-    type: Attribute
+    type: AttributeCovT
```

Comment:
Not sure if this covered further down the stack, but the get method below probably doesn't have tight enough type bounds.

Reply:
I don't think it should actually have a better bound.
If you have an `SSAValue[T] | Operation`, you can only get an `SSAValue` back, because you don't know the type of the operation.
If you have an `SSAValue[T]`, then you wouldn't call `get` in the first place I think?

Reply:
Yeah, on second thoughts I think you are correct.